UnityProject/Assets/Scripts/Systems/Chemistry/Effect.cs

```diff
@@ -6,6 +6,7 @@ namespace Chemistry
 {
 	public abstract class Effect : ScriptableObject
 	{
-		public abstract void Apply(MonoBehaviour sender, float amount);
+		public abstract void Apply(GameObject sender, float amount);
+		public abstract void HeatExposure(GameObject sender, float heat, ReagentMix inMix);
```

Comment:
I see why Alain suggested to make it virtual on discord. Since not every ReactionEffect uses the heat exposure, it is odd having them obliged to implement this. Making it a virtual with an empty body (or perhaps leaving a message stating that the reaction was exposed to heat, but the implementation wasn't overriden) is fair in my opinion.